pkg/app/pipectl/cmd/initialize/kubernetes.go

```diff
@@ -0,0 +1,219 @@
+	switch manifestManager {
+	case kustomize:
+		deploymentInput, err = kustomizeInput(p)
+	case helm:
+		deploymentInput, err = helmInput(p)
+	default:
+		return nil, errors.New("invalid number")
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	spec := &genericKubernetesApplicationSpec{
+		Name:        appName,
+		Input:       *deploymentInput,
+		Description: "Generated by `pipectl init`. See https://pipecd.dev/docs/user-guide/configuration-reference/ for more.",
+	}
+
+	return &genericConfig{
+		Kind:            config.KindKubernetesApp,
+		APIVersion:      config.VersionV1Beta1,
+		ApplicationSpec: spec,
+	}, nil
+}
+
+// Genarate a KubernetesDeploymentInput for Kustomize
+func kustomizeInput(p prompt.Prompt) (*config.KubernetesDeploymentInput, error) {
+	var (
+		kustomizeVersion string
+	)
+	inputs := []prompt.Input{
+		{
+			Message:       "Kustomize version",
+			TargetPointer: &kustomizeVersion,
+			Required:      false,
+		},
+	}
+
+	err := p.RunSlice(inputs)
+	if err != nil {
+		return nil, err
+	}
+
+	deploymentInput := &config.KubernetesDeploymentInput{
+		KustomizeVersion: kustomizeVersion,
+	}
+
+	return deploymentInput, nil
+}
+
+func helmInput(p prompt.Prompt) (*config.KubernetesDeploymentInput, error) {
+	var (
+		helmVersion string
+
+		// remote chart
+		chartRepository string
+		chartName       string
+		chartVersion    string
+		// local chart
+		chartPath string
+
+		// helm options
+		releaseName string
+		valueFiles  []string
+	)
+
+	basicInputs := []prompt.Input{
+		{
+			Message:       "Helm version",
+			TargetPointer: &helmVersion,
+			Required:      false,
+		},
+		{
+			Message:       "Helm Chart Repository (if empty, use local chart)",
+			TargetPointer: &chartRepository,
+			Required:      false,
+		},
+	}
+
+	// If chartRepository is not empty, use remote chart
+	remoteChartInputs := []prompt.Input{
+		{
+			Message:       "Name of the Helm Chart",
+			TargetPointer: &chartName,
+			Required:      true,
+		},
+		{
+			Message:       "Version of the Helm Chart",
+			TargetPointer: &chartVersion,
+			Required:      true,
+		},
+	}
+
+	// If chartRepository is empty, use local chart
+	localChartInputs := []prompt.Input{
+		{
+			Message:       "Relative path from the repository root to the chart directory",
+			TargetPointer: &chartPath,
+			Required:      true,
+		},
+	}
+
+	// Helm options
+	helmOptionsInputs := []prompt.Input{
+		{
+			Message:       "Release name of helm deployment",
+			TargetPointer: &releaseName,
+			Required:      false,
+		},
+		{
+			Message:       "Value files (separated by space)",
+			TargetPointer: &valueFiles,
+			Required:      false,
+		},
+	}
+
+	err := p.RunSlice(basicInputs)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(chartRepository) > 0 {
+		// Use remote chart
+		err = p.RunSlice(remoteChartInputs)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		// Use local chart
+		err = p.RunSlice(localChartInputs)
+		if err != nil {
+			return nil, err
+		}
+	}
```

Comment:
[nit] It would be nice to define the `remoteChartInputs` and `localChartInputs` in each if statement to minimize their scope. 👀